source/server/api_listener_impl.cc

```diff
@@ -32,13 +32,25 @@ void ApiListenerImplBase::SyntheticReadCallbacks::SyntheticConnection::raiseConn
 HttpApiListener::HttpApiListener(const envoy::config::listener::v3::Listener& config,
                                  ListenerManagerImpl& parent, const std::string& name)
     : ApiListenerImplBase(config, parent, name) {
+  TRY_ASSERT_MAIN_THREAD
   auto typed_config = MessageUtil::anyConvertAndValidate<
-      envoy::extensions::filters::network::http_connection_manager::v3::HttpConnectionManager>(
-      config.api_listener().api_listener(), factory_context_.messageValidationVisitor());
+      envoy::extensions::filters::network::http_connection_manager::v3::
+          EnvoyMobileHttpConnectionManager>(config.api_listener().api_listener(),
+                                            factory_context_.messageValidationVisitor());
 
   http_connection_manager_factory_ = Envoy::Extensions::NetworkFilters::HttpConnectionManager::
       HttpConnectionManagerFactory::createHttpConnectionManagerFactoryFromProto(
-          typed_config, factory_context_, read_callbacks_);
+          typed_config.config(), factory_context_, read_callbacks_, false);
+  END_TRY
+  catch (const EnvoyException& e) {
+    auto typed_config = MessageUtil::anyConvertAndValidate<
```

Comment:
nit: this is a bit odd to retry during the exception catch. Can you potentially just match on the type_url of the config and then have a better not supported message if it's neither of the supported ones? Or we could do a specific registry for things that work with the API listener? At minimum I would do a comment/TODO here.

Reply:
would you accept a tracking bug and a follow-up PR?
I'd prefer to make sure this all hangs together on the mobile side (as the first round didn't) and adding a TODO would require another CI cycle.

Reply:
https://github.com/envoyproxy/envoy/issues/17434 filed optimistically but no pressure :-)

Reply:
Sure a follow up is fine, thanks.